lib/Doctrine/Inflector/RulesetInflector.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Inflector;
+
+use Doctrine\Inflector\Rules\Ruleset;
+
+class RulesetInflector implements WordInflector
+{
+    /** @var Ruleset */
+    private $ruleset;
+
+    public function __construct(Ruleset $ruleset)
+    {
+        $this->ruleset = $ruleset;
+    }
+
+    public function inflect(string $word) : string
+    {
+        $inflected = $this->ruleset->getIrregular()->inflect($word);
+
+        if ($inflected !== $word) {
+            return $inflected;
+        }
+
+        if ($this->ruleset->getUninflected()->isUninflected($word)) {
```

Comment:
Imho this check should be prioritized before attempting to inflect an _irregular_.